web/src/lib/domain-logic/essentialQuestions.ts

```diff
@@ -0,0 +1,155 @@
+import { EssentialQuestionsProps, ProfileLabelField } from "@/lib/types/types";
+import {
+  emptyIndustrySpecificData,
+  Industry,
+  IndustrySpecificData,
+  LookupIndustryById,
+  ProfileData,
+} from "@businessnjgovnavigator/shared";
+
+type EssentialQuestionFunction = (industry: Industry) => boolean;
+interface EssentialQuestion
+  extends Omit<EssentialQuestionsProps<ProfileData[keyof ProfileData]>, "fieldName" | "choices"> {
+  fieldName: keyof IndustrySpecificData;
+  validationFunc: EssentialQuestionFunction;
+  reset: boolean;
+  convenienceFunc: (industryId: string | undefined) => boolean;
+}
+
+export const getResetIndustrySpecificData = (industryId: string | undefined) => {
+  const industry = LookupIndustryById(industryId);
+  return EssentialQuestions.filter((eQ) => {
+    return eQ.reset && !eQ.validationFunc(industry);
+  }).reduce((reducer, eQ) => {
+    return { ...reducer, [eQ.fieldName]: emptyIndustrySpecificData[eQ.fieldName] };
+  }, {} as Partial<IndustrySpecificData>);
+};
+
+export const getValidationFuncByFieldName = (fieldName: ProfileLabelField) => {
```

Comment:
I didn't realize that we didn't have the eslint rule that enforces return types on functions, but we should add it.